### PR TITLE
Implement zipWith

### PR DESCRIPTION
**Pull request by vascoorey**
Implemented zipWith and added test cases.

In the case of array size mis-match it will zip up to the smallest array size.

### CODE REVIEW

File: Underscore/USArrayWrapper.m [hunk 1]
```diff
@@ -225,6 +225,26 @@ - (USArrayWrapper *)shuffle
     };
 }
 
+- (USArrayWrapper *(^)(NSArray *array, UnderscoreArrayZipWithBlock block))zipWith
+{
+    return ^USArrayWrapper *(NSArray *array, UnderscoreArrayZipWithBlock block) {
+        NSMutableArray *result = [NSMutableArray arrayWithCapacity:self.array.count];
+        if(self.array.count <= array.count)
+        {
```
Comment (robb) [inline Underscore/USArrayWrapper.m]: minor style issue: can you put that one on the same line as the if?

File: Underscore/USArrayWrapper.m [hunk 2]
```diff
@@ -225,6 +225,26 @@ - (USArrayWrapper *)shuffle
...
+            [self.array enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
+                [result addObject:block(obj, array[idx])];
+            }];
+        }
+        else
+        {
```
Comment (robb) [inline Underscore/USArrayWrapper.m]: can you make that `} else {` ?

**Comment by robb** [comment]
You're on a roll! :+1:  Two minor comments & please add yourself to the CONTRIBUTORS file if you'd like

**Comment by vascoorey** [comment]
Roger that ;)

**Comment by vascoorey** [comment]
Done!

**Comment by robb** [comment]
:sparkling_heart:

**Pull request merged**